cvasl/seperated.py

```diff
@@ -537,3 +550,200 @@ def check_sex_dimorph_expectations(dataframe):
+    and raw y_test versus y_pred
+    """
+    y_split = our_ml_matrix['sex'].values
+# 5 folds as example, you can change this
+    sss = StratifiedShuffleSplit(n_splits=5, test_size=0.25, random_state=12)
+    X = our_x
+    y = our_y
+    sss.get_n_splits(X, y_split)
+    print(sss)
+    print(f'Whole dataset shape: X {X.shape}, y {y.shape}')
+    unique, counts = np.unique(y_split, return_counts=True)
+    print(f'Classes: {unique}, percentages: {100*counts/y.shape[0]}')
+    y_frame = []
+    all_mod_results = []
+    models = []
+    for i, (train_index, test_index) in enumerate(sss.split(X, y_split)):
+        print(f"\nFold {i}:")
+        print(
+            f'Train shapes: X {X[train_index].shape}',
+            f' y {y[train_index].shape}'
+        )
+        unique, counts = np.unique(y_split[train_index], return_counts=True)
+        print(
+            f'Sex classes: {unique}',
+            f'percentages: {100*counts/y[train_index].shape[0]}'
+        )
+        print(
+            f'\nTest shapes: X {X[test_index].shape}',
+            f'  y {y[test_index].shape}'
+        )
+        unique, counts = np.unique(y_split[test_index], return_counts=True)
+        print(
+            f'Sex classes: {unique},'
+            f'percentages: {100*counts/y[test_index].shape[0]}'
+        )
+        cols = [
+            'algorithm',
+            'fold',
+            'file_name',
+            'mae',
+            'r2',
+            'explained_variance']
+        mod_results = pd.DataFrame(columns=cols)
+        current_fold_X_train = X[train_index]
+        current_fold_y_train = y[train_index]
+        current_fold_X_test = X[test_index]
+        current_fold_y_test = y[test_index]
+        scikit_model.fit(current_fold_X_train, current_fold_y_train)
+        current_fold_y_pred = scikit_model.predict(current_fold_X_test)
+
+        data = [[
+            model_name + str(i),
+            i,
+            model_file_name + str(i) + '.sav',
+            mean_absolute_error(current_fold_y_test, current_fold_y_pred),
+            scikit_model.score(current_fold_X_test, current_fold_y_test),
+            metrics.explained_variance_score(
+                current_fold_y_test,
+                current_fold_y_pred)]]
+        mod_results_current_fold = pd.DataFrame(data, columns=cols)
+        mod_results = pd.concat([mod_results, mod_results_current_fold])
+        mod_results.reset_index(drop=True, inplace=True)
+        all_mod_results.append(mod_results)
+        y_frame_now = pd.DataFrame(
+            {'y_test': list(current_fold_y_test),
+             'y_pred': list(current_fold_y_pred)})
+
+        y_frame.append(y_frame_now)
+        models.append(scikit_model)
+
+    df = pd.concat(all_mod_results)
+    y_frame = pd.concat([
+        y_frame[0],
+        y_frame[1],
+        y_frame[2],
+        y_frame[3],
+        y_frame[4],], axis=0)
+
+    return df, y_frame, models
+
+
+def frame_a_model_sex_split_2(
```

Comment:
I think this function is exactly the same as `frame_a_model_sex_split`, but doesn't contain the print statements. 

The comments I left for the other function are then valid for this one as well. I'd suggest to keep only one between these two. If you want to make the print statements optional, you could always add a boolean variable as input for controlling the print vs no print

Reply:
New more generalizable function is  stratified_one_category_shuffle_split